### PR TITLE
Make snapshot support and auto release version update

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
tendrl-bug-id: Tendrl/api#209
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@r0h4n , @anivargi , @mbukatov , @dahorak , please review

File: Makefile [hunk 1]
```diff
@@ -40,3 +39,19 @@ rpm:
...
 	fi
+
+srpm:
+	@echo "target: rpm"
+	@echo  "  ...building rpm $(V_ARCH)..."
+	rm -fr $(BUILDS)
+	mkdir -p $(DEPLOY)/latest
+	mkdir -p $(RPMBUILD)/SPECS
+	sed -e "s/@VERSION@/$(VERSION)/" tendrl-api.spec \
+	        > $(RPMBUILD)/SPECS/tendrl-api.spec
```
Comment (mbukatov) [inline Makefile]: Here you replace string `@VERSION@` with value from `VERSION` variable, but there is no such string anywhere but in the `Makefile`:

```
$ git grep '@VERSION@'                                                         
Makefile:       sed -e "s/@VERSION@/$(VERSION)/" tendrl-api.spec \             
Makefile:       sed -e "s/@VERSION@/$(VERSION)/" tendrl-api.spec \  
```

File: Makefile [hunk 2]
```diff
@@ -40,3 +39,19 @@ rpm:
 		printf "\nThe tendrl-api RPMs are located at:\n\n"; \
 		printf "   $(DEPLOY)/latest\n\n\n\n"; \
 	fi
+
+srpm:
```
Comment (mbukatov) [inline Makefile]: Makefile target `srpm` requires tarball, which is created during `dist` target.

**Comment by mbukatov** [comment]
I'm going to check makefiles of other repositories, and hopefully suggest some changes as it would be nice to unify the makefiles and rpmbuild phase for all repositories as much as possible.

**Comment by mbukatov** [comment]
Here is my proposal how to fix the issues, based on specfile of tendrl-dashboard (now tendrl-ui): https://github.com/mbukatov/api/commit/352666ab340e83f4ef4f052644bf775d8c576358

If you like it, you can use my commit in your request like this (assuming you are in tendrl-api repository with auto-release-version branch checked out):

```
$ git remote add mbukatov git@github.com:mbukatov/api.git
$ git fetch mbukatov
$ git cherry-pick 352666ab340e83f4ef4f052644bf775d8c57635
```

**Comment by TimothyAsir** [comment]
@mbukatov please review

**Pull request merged**